src/Controller/ResetPasswordControllerTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SymfonyCasts\Bundle\ResetPassword\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Provides useful methods to a "reset password controller"
+ *
+ * Use of this trait requires a controller to extend
+ * Symfony\Bundle\FrameworkBundle\Controller\AbstractController
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+trait ResetPasswordControllerTrait
+{
+    private function setCanCheckEmailInSession(): void
+    {
+        $this->getSession()->set('ResetPasswordCheckEmail', true);
+    }
+
+    private function isAbleToCheckEmail(): bool
```

Comment:
```suggestion
    private function canCheckEmail(bool $removeFromSession = true): bool
```
2 things:

1) The "can" matches the other method
2) The argument makes it more flexible... but I'm *really* adding it because it will at least make it a *bit* more obvious that this method will *also* remove the value from the session. I wish we could fit that into the name somehow - but I can't find anything reasonable - e.g. `canCheckEmailAndClearSession()` or something :p. The other option would be to split into 2 separate methods - it would require one more line in the controller - e.g. `$this->clearCheckEmailSessionFlag()` or something, but would be more clear (and maybe the "set"` is `$this->setCheckEmailSessionFlag()`?)

Hmm, actually, won't we have the same problem with the "token"? We would want to remove that from the session after it's used? Regardless of how we do it (clear session or something else), we just need to make sure that the page will 404 after the token is used.

Reply:
1 - done

2 - added another method to the trait `cleanSessionAfterReset()`. instead of clearing the email flag from the session after viewing the check email page, it will be done after a request has been removed from persistence (after validation).. this will allow us to clear both the email flag and public token in one shot..

we could still split these 2 removal methods up if need be..

Reply:
WIP Example of a ResetPasswordController https://gist.github.com/jrushlow/59fc19d6558ef67006e2d23881ebac8a